File: src/main/java/dev/klomptech/jbark/interpreter/PrintStyles.java
package dev.klomptech.jbark.interpreter;

import dev.klomptech.jbark.print.PrintStyle;

// Formats output differently per print verb
public final class PrintStyles {

  private PrintStyles() {}

  public static String format(final PrintStyle style, final String text) {
    PrintStyle resolvedStyle = style != null ? style : PrintStyle.BARK;
    String safeText = text != null ? text : "";
    return switch (resolvedStyle) {
      case BARK -> safeText;
      case GROWL, WOOF -> safeText.toUpperCase();
      case HOWL -> safeText.isEmpty() ? safeText : (safeText.charAt(0) + "").repeat(4) + safeText;
      case WHIMPER, WHINE -> safeText.toLowerCase();
      case YAPPING -> (safeText + " ").repeat(2).trim();
      case WHINING -> String.join(" ", safeText.split(""));
    };
  }

  // Default word when a print verb has no value. Formatted by format().
  public static String bareText(final PrintStyle style) {
    PrintStyle resolvedStyle = style != null ? style : PrintStyle.BARK;
    return switch (resolvedStyle) {
      case BARK, WOOF -> "woof";
      case GROWL -> "growl";
      case HOWL -> "howl";
      case WHIMPER -> "whimper";
      case WHINE, WHINING -> "whine";
      case YAPPING -> "yap";
    };
  }
}

File: pbark/pbark/interpreter/print_styles.py
from __future__ import annotations

from pbark.interpreter import bark_value as bv
from pbark.print_style import PrintStyle


class PrintStyles:
    @staticmethod
    def format(style: PrintStyle | None, text: str | None) -> str:
        resolved = style if style is not None else PrintStyle.BARK
        safe = text if text is not None else ""
        if resolved in (PrintStyle.BARK,):
            return safe
        if resolved in (PrintStyle.GROWL, PrintStyle.WOOF):
            return safe.upper()
        if resolved is PrintStyle.HOWL:
            return safe if not safe else safe[0] * 4 + safe
        if resolved in (PrintStyle.WHIMPER, PrintStyle.WHINE):
            return safe.lower()
        if resolved is PrintStyle.YAPPING:
            return ((safe + " ") * 2).strip() if safe else ""
        if resolved is PrintStyle.WHINING:
            return " ".join(list(safe))
        return safe

    @staticmethod
    def bare_text(style: PrintStyle | None) -> str:
        resolved = style if style is not None else PrintStyle.BARK
        mapping = {
            PrintStyle.BARK: "woof",
            PrintStyle.WOOF: "woof",
            PrintStyle.GROWL: "growl",
            PrintStyle.HOWL: "howl",
            PrintStyle.WHIMPER: "whimper",
            PrintStyle.WHINE: "whine",
            PrintStyle.WHINING: "whine",
            PrintStyle.YAPPING: "yap",
        }
        return mapping.get(resolved, "woof")
